### ISSUE TITLE
Prepare Kafka integration tests

### ISSUE THREAD

**Issue by fkaltner** [eclipse/hono#2469]
In parallel to the ongoing Kafka migration tasks the Hono integration tests need to spin up a Kafka (and Zookeeper) container which is properly configured for Docker-internal access (by Protocol Adapters and Device Registries) and Docker-external access (by the integration tests themselves)

Furthermore the integration test helpers need to be adapted to use the currently developed _org.eclipse.hono.application.client.ApplicationClientFactory_ interface.

In addition to that we need to make up our minds about the test concept itself considering the fact that it is likely that Hono will support AMQP and Kafka in parallel based on a configuration property of a tenant. Strictly considered this means that we need to run every test once with AMQP and once with Kafka. This will obviously at least double the runtime of our tests.

**Comment by fkaltner** [comment]
@sophokles73 may I work on this?

**Comment by fkaltner** [comment]
@b-abel Here's a draft PR which can be used to play around https://github.com/eclipse/hono/pull/2473

Note that the integration tests will still fail as northbound consumers still connect to AMQP. I'll continue to work on that and update the PR as soon as I am done.

Regarding the device registries: will you update the configuration of those similar to _AbstractAdapterConfig_, so that Kafka can be enabled by configuration properties?

**Comment by b-abel** [comment]
> @b-abel Here's a draft PR which can be used to play around #2473

Cool. I'll have a look.

> Regarding the device registries: will you update the configuration of those similar to _AbstractAdapterConfig_, so that Kafka can be enabled by configuration properties?

This is currently not on my priority list :smile:  Could you please create an issue for it?

**Comment by fkaltner** [comment]
> This is currently not on my priority list 😄 Could you please create an issue for it?

Sure. Created #2475.

**Comment by sophokles73** [comment]
@fkaltner can this be closed?

**Comment by fkaltner** [comment]
Integration tests for request/response commands are still not there as the implementation is not complete yet. But we can also create a new issue for that as this issue has the 1.7.0. milestone.

**Comment by sophokles73** [comment]
Yes, that is what I would propose as well. Let's close this issue as it has indeed laid out the groundwork necessary for implementing the integration tests. can you create a new issue for the request/response command tests and then close this issue?
